src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -352,10 +352,59 @@ public StudyEntity doGetStudyAndUpdateIsPrivate(UUID studyUuid, String headerUse
         return Mono.fromCallable(() -> studyInfosService.search(query)).flatMapMany(Flux::fromIterable);
     }
 
-    Flux<EquipmentInfos> searchEquipments(@NonNull UUID studyUuid, @NonNull String query) {
+    public static String escapeLucene(String s) {
+        StringBuilder sb = new StringBuilder();
+
+        for (int i = 0; i < s.length(); ++i) {
+            char c = s.charAt(i);
+            switch (c) {
+                case '+':
+                case '\\':
+                case '-':
+                case '!':
+                case '(':
+                case ')':
+                case ':':
+                case '^':
+                case '[':
+                case ']':
+                case '"':
+                case '{':
+                case '}':
+                case '~':
+                case '*':
+                case '?':
+                case '|':
+                case '&':
+                case '/':
+
```

Comment:
why this blank line ? it seems that there is something missing

Reply:
I wanted to outstand the white space from the other characters, as this is hte wanted difference from the Lucene escape.